ipa-core/src/protocol/basics/reveal.rs

```diff
@@ -50,26 +52,33 @@ pub trait Reveal<C: Context, B: RecordBinding>: Sized {
 /// i.e. their own shares and received share.
 #[async_trait]
 #[embed_doc_image("reveal", "images/reveal.png")]
-impl<C: Context, V: SharedValue> Reveal<C, RecordId> for Replicated<V> {
-    type Output = V;
+impl<C: Context, V: SharedValue + Vectorizable<N>, const N: usize> Reveal<C, N>
+    for Replicated<V, N>
+{
+    type Output = <V as Vectorizable<N>>::Array;
 
-    async fn reveal<'fut>(&self, ctx: C, record_id: RecordId) -> Result<V, Error>
+    async fn reveal<'fut>(
```

Comment:
want to try your `Cow` trick for RPITIT here? we can always do it later though

Reply:
I ended up with something [a bit different from the `Cow` solution](https://github.com/private-attribution/ipa/blob/57b4d4952534bdcd0c7d481a5d070596486b5068/ipa-core/src/protocol/basics/mul/mod.rs#L79).

I'm still not confident that either solution is reasonable to use 100% of the time, but I can try to use it here. I'd rather do it as a follow-up though.